### PR TITLE
feat: Update website input to accept more kinds of URLs

### LINKED ISSUE #1922: Website Input Format Is Too Demanding
When attempting to update the website field on the profile details page, the website input requires a user to input `http://www` in order to be accepted. This creates an obstacle that's too demanding for users.

Screenshot of the Profile Details page showing that google.com is considered an invalid website

Appropriate sanitization should be applied on this input to prevent JavaScript injections. Appropriate whitespace trimming should also be applied. 

**To-do**

* Update the rules for a valid website in `profileMetadata.js`

**Review Checklist**

* `google.com` is accepted as a valid URL in the website field
* Necessary backend unit testing has been added
* Necessary frontend unit testing has been added
* Necessary integration testing has been added
* Documentation has been updated to reflect new changes

### PR DESCRIPTION
**Pull request by gmeben**
This pull request will make the website input on the Profile page allow the following URLs to be added to a user's profile:

```
google.com
google.com/
google.com/about
http://google.com
http://google.com/
http://google.com/about
https://google.com
https://google.com/
https://google.com/about
```

Sanitization and validation have been added on the `websites` field in `profile_metadata`. 

Note: whitespace trimming already takes place on the client and the backend for this input.

Closes #1922

### CODE REVIEW

**Comment by gmeben** [comment]
As advised, I'm going to try to refactor this so that it no longer utilizes a regex.

**Comment by gmeben** [comment]
Unfortunately, a regex is still necessary here but a new (and hopefully more efficient) regex has been added.

File: backend/graphql/schema.graphql [hunk 1]
```diff
@@ -201,5 +201,5 @@ input UpdateProfileMetadataInput @validator(class: "ProfileMetadataValidator") {
     affiliation: String
     interest_keywords: [String]
     biography: String
-    websites: [String]
+    websites: [String] @website
```
Comment (wreality) [inline backend/graphql/schema.graphql:204]:
```suggestion
    websites: [String] @sanitizeURL
```
Just suggesting so its clearer what the effect of the directive is.

**Review by wreality — APPROVED** [review]
Looks good!

**Pull request merged**